target/linux/ramips/dts/mt7621_asus_rt-n56u-b1.dts

```diff
@@ -0,0 +1,190 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "asus,rt-n56u-b1", "mediatek,mt7621-soc";
+	model = "ASUS RT-N57U B1";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "blue:power";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "blue:wlan2g";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0radio";
+		};
+
+		wlan5g {
+			label = "blue:wlan5g";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1radio";
+		};
+
+		wan {
+			label = "blue:wan";
+			gpios = <&gpio 7 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "dsa-0.0:04:link";
+		};
+
+		lan {
+			label = "blue:lan";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "dsa-0.0:03:link";
+		};
+
+		usb {
+			label = "blue:usb";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+			trigger-sources = <&xhci_ehci_port1>, <&ehci_port2>;
+			linux,default-trigger = "usbport";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 18 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_WPS_BUTTON>;
+			debounce-interval = <60>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "config";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0xfb0000>;
+			};
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+
+		led {
+			led-sources = <2>;
+			led-active-low;
+		};
+	};
+};
+
+&pcie1 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+
+		led {
+			led-active-low;
+		};
+	};
+};
+
+&gmac0 {
+	mtd-mac-address = <&factory 0x0022>;
```

Comment:
Please just use 0x22 without the leading zeros. That location is relatively uncommon; are you sure?

Reply:
Ok. I understand this now.

Reply:
Asking again: Are you sure that 0x22 contains a valid MAC address and that this one is the correct one for LAN?

Reply:
This Asus router have the following MAC addresses offset:
```
0x4 - MAC 2.4GHz
0x8004 - MAC 5GHz
0x22 - MAC have same value as 5G
0x28 - MAC have same value as 2.4G
```
I check it in my factory dump.
WAN=2.4GHz and LAN=5GHz
I look a asus-merlin sources and find:
```
#elif defined(RTAC52U) || defined(RTAC51U) || defined(RTN54U) || defined(RTAC54U) || defined(RTAC1200HP) || defined(RTN56UB1) || defined(RTN56UB2)
#define OFFSET_MAC_ADDR_2G	0x40004
#define OFFSET_MAC_ADDR		0x48004
#define OFFSET_MAC_GMAC0	0x40022
#define OFFSET_MAC_GMAC2	0x40028
```
Ok. I make a some investigation. 
I download to router factory firmware and change hex value MAC addresses in factory image.
Stock firmware don't used adresses in offset 0x22 and 0x28. But in this addresses really true value of MAC.
Stock firmware using only 0x4 and 0x8004 from radio interfaces MAC. 
Ok. I can change DT to this value.

Reply:
> Stock firmware don't used adresses in offset 0x22 and 0x28. But in this addresses really true value of MAC.
> Stock firmware using only 0x4 and 0x8004 from radio interfaces MAC.

So, the OFFSET_MAC_GMAC0 and ..._GMAC2 pasted above are never used?

Reply:
I put my own values ​​in these cells and did not see that they are used with the factory firmware.
The factory firmware uses values ​​from: OFFSET_MAC_ADDR_2G - as WAN, OFFSET_MAC_ADDR - as LAN

Reply:
I checked these values ​​only on my device in the factory dump.
The factory firmware uses the values ​​from the Wi-Fi MAC offset. In SDK from Merlin(fork from Asus SDK) 0x22 and 0x28 are indicated. Yes, in these cells I have the correct values in my factory dump, but not the fact that others will.

Reply:
Now i have something like this:
```
br-lan    Link encap:Ethernet  HWaddr 10:7B:44:55:66:9C  
eth0      Link encap:Ethernet  HWaddr 10:7B:44:55:66:9C  
lan1      Link encap:Ethernet  HWaddr 10:7B:44:55:66:9C  
lan2      Link encap:Ethernet  HWaddr 10:7B:44:55:66:9C  
lan3      Link encap:Ethernet  HWaddr 10:7B:44:55:66:9C  
lan4      Link encap:Ethernet  HWaddr 10:7B:44:55:66:9C  
wan       Link encap:Ethernet  HWaddr 10:7B:44:55:66:98  
wlan0     Link encap:Ethernet  HWaddr 10:7B:44:55:66:98
wlan1     Link encap:Ethernet  HWaddr 10:7B:44:55:66:9C 
```